File: app/src/main/java/com/gmail/fattazzo/formula1world/ergast/json/objects/Driver.java
package com.gmail.fattazzo.formula1world.ergast.json.objects;

import com.gmail.fattazzo.formula1world.domain.F1Driver;

import org.apache.commons.lang3.time.DateUtils;

import java.io.Serializable;

/**
 * @author fattazzo
 *         <p/>
 *         date: 06/06/17
 */
public class Driver implements Serializable {

    private String driverId;
    private int permanentNumber;
    private String code;
    private String url;
    private String givenName;
    private String familyName;
    private String dateOfBirth;
    private String nationality;

    public Driver(String driverId, int permanentNumber, String code, String url, String givenName, String familyName, String dateOfBirth, String nationality) {
        this.driverId = driverId;
        this.permanentNumber = permanentNumber;
        this.code = code;
        this.url = url;
        this.givenName = givenName;
        this.familyName = familyName;
        this.dateOfBirth = dateOfBirth;
        this.nationality = nationality;
    }

    public F1Driver toF1Driver() {
        F1Driver f1Driver = new com.gmail.fattazzo.formula1world.domain.F1Driver();
        f1Driver.driverRef = this.driverId;
        f1Driver.code = this.code;
        try {
            f1Driver.dateOfBirth = DateUtils.parseDate(this.dateOfBirth, "yyyy-MM-dd");
        } catch (Exception e) {
            f1Driver.dateOfBirth = null;
        }
        f1Driver.familyName = this.familyName;
        f1Driver.givenName = this.givenName;
        f1Driver.number = this.permanentNumber;
        f1Driver.url = this.url;
        f1Driver.nationality = this.nationality;

        return f1Driver;
    }

    @Override
    public String toString() {
        return "Driver{" +
                "driverId='" + driverId + '\'' +
                ", permanentNumber=" + permanentNumber +
                ", code='" + code + '\'' +
                ", url='" + url + '\'' +
                ", givenName='" + givenName + '\'' +
                ", familyName='" + familyName + '\'' +
                ", dateOfBirth='" + dateOfBirth + '\'' +
                ", nationality='" + nationality + '\'' +
                '}';
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/ergast/json/objects/Driver.kt
/*
 * Project: total-gp-world
 * File: Driver.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.ergast.json.objects

import com.gmail.fattazzo.formula1world.domain.F1Driver

import org.apache.commons.lang3.time.DateUtils

import java.io.Serializable

/**
 * @author fattazzo
 *
 *
 * date: 06/06/17
 */
class Driver(private val driverId: String, private val permanentNumber: Int, private val code: String, private val url: String, private val givenName: String, private val familyName: String, private val dateOfBirth: String, private val nationality: String) : Serializable {

    fun toF1Driver(): F1Driver {
        val f1Driver = com.gmail.fattazzo.formula1world.domain.F1Driver()
        f1Driver.driverRef = this.driverId
        f1Driver.code = this.code
        try {
            f1Driver.dateOfBirth = DateUtils.parseDate(this.dateOfBirth, "yyyy-MM-dd")
        } catch (e: Exception) {
            f1Driver.dateOfBirth = null
        }

        f1Driver.familyName = this.familyName
        f1Driver.givenName = this.givenName
        f1Driver.number = this.permanentNumber
        f1Driver.url = this.url
        f1Driver.nationality = this.nationality

        return f1Driver
    }

    override fun toString(): String {
        return "Driver{" +
                "driverId='" + driverId + '\'' +
                ", permanentNumber=" + permanentNumber +
                ", code='" + code + '\'' +
                ", url='" + url + '\'' +
                ", givenName='" + givenName + '\'' +
                ", familyName='" + familyName + '\'' +
                ", dateOfBirth='" + dateOfBirth + '\'' +
                ", nationality='" + nationality + '\'' +
                '}'
    }
}
